sdk/core/azure-core-amqp/inc/azure/core/amqp/internal/message_sender.hpp

```diff
@@ -133,13 +141,16 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
     /** @brief Opens a message sender.
      *
      * @param context The context to use for the operation.
+     *
+     * @return An error if the operation was not successful.
      */
-    void Open(Context const& context = {});
+    _azure_NODISCARD Models::_internal::AmqpError Open(Context const& context = {});
 
     /** @brief Closes a message sender.
      *
+     * @return An error if the operation was not successful.
      */
-    void Close(Context const& context = {});
+    Models::_internal::AmqpError Close(Context const& context = {});
```

Comment:
Why is unacceptable to discard the open but ok to discard the Close?

Reply:
`Open` can fail.
`Close` cannot fail.

That's a really good point. Changing it back to void() and throwing an exception on failure.